src/rsz/src/RepairDesign.cc

```diff
@@ -746,16 +776,20 @@ RepairDesign::repairNetWire(BufferedNetPtr bnet,
         - max_load_slew_margined / elmore_skew_factor_;
       float l = (-b + sqrt(b*b - 4 * a * c)) / (2 * a);
       if (l >= 0.0) {
-        split_length = min(split_length, metersToDbu(l));
+        if (split_length > 0.0)
+          split_length = min(split_length, metersToDbu(l));
+        else
+          split_length = metersToDbu(l);
         split_wire = true;
         resize = false;
       }
       else {
         split_length = 0;
-        split_wire = true;
-        resize = false;
+        split_wire = false;
+        resize = true;
```

Comment:
Why the change here?

Reply:
This looks to be a bug if you look at the code in it's wider context.

Reply:
I feel like we should pull this change into its own PR since it seems unrelated to the infinite loop bug. Unless I'm mistaken.